initial cli structure proposal

ryankwilliams opened the pull request:
This commit contains a proposed structure for the cli. Each REST
collection has its own class with methods for its available actions.

ryankwilliams committed:
initial cli structure proposal

This commit contains a proposed structure for the cli. Each REST
collection has its own class with methods for its available actions.

tehsmyers reviewed (changes requested):
I'm not sure if this is what we want to call it, but we should expose the entry point as a CLI script:
```diff
diff --git a/setup.cfg b/setup.cfg
index 61fe184..58955de 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -32,6 +32,10 @@ keywords =
 packages =
     miqcli
 
+[entry_points]
+console_scripts =
+    miqcli = miqcli.cli.main:cli
+
 [wheel]
 universal = 1
 
```

After adding that and installing with `pip install -e .`, the miqcli command fails with:

```
$ miqcli
Traceback (most recent call last):
  File "/home/semyers/ve/miqcli/bin/miqcli", line 6, in <module>
    from miqcli.cli.main import cli
  File "/home/semyers/ve/miqcli/manageiq-cli/miqcli/cli/main.py", line 4, in <module>
    from xmlrpclib import ServerProxy
ModuleNotFoundError: No module named 'xmlrpclib'
```

We claim to support python >= 2.7 in setup.cfg, so the rename of xmlrpclib to xmlrpc.client should be handled.

ryankwilliams commented:
@seandst Thank you for the comments. The name can be changed at anytime. I know @tvieira put that name as a base when creating the initial project structure.

I will amend to the PR to fix the failed import for Python >= 2.7 support and update the setup.cfg to expose the CLI script as an entrypoint.

ryankwilliams committed:
handle xmlrpclib rename for python >= 2.7 compatibility

ryankwilliams committed:
define entry points section within setup.cfg

Reverted the setup.py back to its normal state to maintain the minimal
data required by pbr. Defines the console script to be executed within
the setup.cfg following pbr standards.

tvieira commented on miqcli/cli/main.py:
```diff
@@ -1,7 +1,11 @@
+try:
+    from xmlrpclib import ServerProxy
+except ImportError:
+    from xmlrpc.client import ServerProxy
```
We will probably have to handle few other incompatibilities between python versions. Wouldn't be better to use a _compat module within the main package to handle this? My suggestion, but I would be fine with this for now too.

ryankwilliams replied:
I am in favor also of having a compat module to handle incompatibilities between python versions. I think once more implementation is in place for the CLI, we can add that module in. Which we could then update the cli.main module to remove this handling of imports.

tehsmyers reviewed (approved):
Stubtastic!
